File: Find Number.cpp
class Solution{
public:
   
    long long findNumber(long long N){
      
      long long a = 2*N-1;
      //cout<<a<<endl;
      string s = to_string(a);
      string ans = s;
      long long  i = s.length()-1;
      int j=1;
      while(i != 0)
      {
          string part = s.substr(0,i);
          string next = s.substr(i);
          long long sum=0;
          sum=stoll(part);
          sum+=(sum-1);
          part = to_string(sum);
          s = part+next;
        ++j;
            i=s.length()-j;
      }
      return stoll(s);
      
    }
};

File: Find Number.java
class Solution
{
     public long findNumber(long N)
    {
        long[] v = { 9, 1, 3, 5, 7 };
        long ans = 0;
        while (N > 0) {
          ans = ans * 10 + v[(int)(N % 5)];
          if (N % 5 == 0) {
            N = N / 5;
            N--;
          }
          else
            N /= 5;
        }
        Queue<Long> q = new ArrayDeque<>();
        while (ans > 0) {
          q.add(ans % 10);
          ans /= 10;
        }
        while (!q.isEmpty()) {
          ans = ans * 10 + q.peek();
          q.remove();
        }
        return ans;
    }
}
